lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -254,6 +254,32 @@ public void LoadClassCache(Type objType, Schema s)
                     break;
                 case NamedSchema ns:
                     EnumCache.AddEnumNameMapItem(ns, objType);
+                    break;
+                case UnionSchema us:
+                    if (us.Schemas.Count == 2 && (us.Schemas[0].Tag == Schema.Type.Null || us.Schemas[1].Tag == Schema.Type.Null) && objType.IsClass)
+                    {
+                        // in this case objType will match the non null type in the union
+                        foreach (var o in us.Schemas)
+                        {
+                            if (o.Tag != Schema.Type.Null)
+                            {
+                                LoadClassCache(objType, o);
+                            }
+                        }
+
+                    }
+                    else
+                    {
+                        // check the schema types are registered
+                        foreach (var o in us.Schemas)
+                        {
+                            if (o.Tag == Schema.Type.Record && GetClass(o as RecordSchema) == null)
+                            {
+                                throw new AvroException($"Class for union record type {o.Fullname} is not registered. Create a ClassCache object and call RegisterUnionTypes");
```

Comment:
`RegisterUnionTypes` isn't a method in `ClassCache`. Did you mean `LoadClassCache`?